cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java

```diff
@@ -396,7 +456,12 @@ public StorageMetadata put(NewArtifact newArtifact, InputStream source)
             }
 
             // Set contentChecksum file attribute
-            setFileAttribute(path, CHECKSUM_ATTRIBUTE_NAME, checksum.toString());
+            setFileAttribute(txnTarget, CHECKSUM_ATTRIBUTE_NAME, checksum.toString());
+
+            // to atomic copy into content directory
+            Path newCopy = Files.move(txnTarget, contentPath, StandardCopyOption.ATOMIC_MOVE);
+            log.debug("moved file to : " + contentPath);
```

Comment:
You also want REPLACE_EXISTING in the options to move. The previous code overwrote the existing file content and only used move to "rollback"... now this code relies on move to replace.

Reply:
ok